client/components/external-link/index.jsx

```diff
@@ -36,7 +36,7 @@ export default React.createClass( {
 			'has-icon': !! this.props.icon,
 		} );
 
-		const props = assign( {}, this.props, {
+		const props = assign( {}, omit( this.props, 'icon', 'iconSize' ), {
```

Comment:
You'd rather whitelist all valid `<a/>` props? I think that's a bit excessive in this case. Once we clear up all existing React errors, you should be able to see pretty quickly if you have passed in an invalid property.